### PR TITLE
Add more code comments to various ECC logic

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Some more code comments, explaining efficiency of batch normalization for curve points in the relevant code comments. Inspired by this coming up as a change in code re #157 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests - n/a
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md` - n/a
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: ec/src/models/short_weierstrass_jacobian.rs [hunk 1]
```diff
@@ -293,13 +293,13 @@ impl<P: Parameters> PartialEq for GroupProjective<P> {
         // The points (X, Y, Z) and (X', Y', Z')
         // are equal when (X * Z^2) = (X' * Z'^2)
         // and (Y * Z^3) = (Y' * Z'^3).
-        let z1 = self.z.square();
-        let z2 = other.z.square();
+        let z1z1 = self.z.square();
```
Comment (ValarDragon) [inline ec/src/models/short_weierstrass_jacobian.rs:296]: Drive by change to make this more consistent with the rest of the SW arithmetic code

File: ec/src/models/short_weierstrass_jacobian.rs [hunk 2]
```diff
@@ -399,6 +399,12 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
 
     #[inline]
     fn batch_normalization(v: &mut [Self]) {
+        // A projective curve element (x, y, z) is normalized
+        // to its affine representation, by the conversion
+        // (x, y, z) -> (x / z^2, y / z^3, 1)
+        // Batch normalizing N short-weierstrass curve elements costs:
+        //     1 inversion + 6N multiplications + N squarings    (Field ops)
+        // With the batch inversion taking 3N of the multiplications
```
Comment (Pratyush) [inline ec/src/models/short_weierstrass_jacobian.rs:407]:
```suggestion
        //     1 inversion + 6N field multiplications + N field squarings    (Field ops)
        // (batch inversion requires 3N multiplications + 1 inversion)
```

File: ec/src/models/twisted_edwards_extended.rs [hunk 3]
```diff
@@ -444,6 +444,12 @@ impl<P: Parameters> ProjectiveCurve for GroupProjective<P> {
     }
 
     fn batch_normalization(v: &mut [Self]) {
+        // A projective curve element (x, y, t, z) is normalized
+        // to its affine representation, by the conversion
+        // (x, y, t, z) -> (x/z, y/z, t/z, 1)
+        // Batch normalizing N twisted edwards curve elements costs:
+        //     1 inversion + 6N multiplications    (Field ops)
+        // With the batch inversion taking 3N of the multiplications
```
Comment (Pratyush) [inline ec/src/models/twisted_edwards_extended.rs:452]:
```suggestion
        //     1 inversion + 6N field multiplications
        // (batch inversion requires 3N multiplications + 1 inversion)
```

**Review by Pratyush — CHANGES_REQUESTED** [review]
Thanks for this! It LGTM modulo the small nits

**Review by Pratyush — APPROVED** [review]

**Pull request merged**